src/main/java/com/uid2/shared/attest/UidCoreClient.java

```diff
@@ -108,6 +106,9 @@ private InputStream getWithAttest(String path, String jwtToken) throws IOExcepti
         }
 
         String attestationToken = attestationTokenRetriever.getAttestationToken();
+        if (jwtToken == null || jwtToken.isEmpty()) {
```

Comment:
Would it be better to put this just after `attestationTokenRetriever.attest();`? `jwtToken` would only really be null if we haven't attested yet

i.e.

```
        if (!attestationTokenRetriever.attested()) {
            attestationTokenRetriever.attest();
            jwtToken = this.getJWT();
        }
```

Reply:
Moved the code, but still left the check. There are other reasons we might not have attested, such as core being unavailable

Reply:
Removed the parameter and call getJWT just before we need it